Wrong expected earnings calculation in backend unavailability (Halvings)

When OCM is calculating expected earnings, the backend server is probed for difficulty and tipheight. If the backend server successfully responds with difficulty and fails to respond with the current tipheight, the earnings calculation will not take the halvings into account. Thus reporting, at this time, an earnings calculation 4x higher. 

Error message OCM gives when this happens:
`2024/01/15 19:05:39.239222 [ERROR] Error fetching tip height: Get "https://ocm-backend.blkidx.org/info": context deadline exceeded (Client.Timeout exceeded while awaiting headers)`

This affects the latest version (v2.2 beta1), and previous versions.

--- a/backend/mining.go
+++ b/backend/mining.go
@@ -79,6 +79,7 @@ func (m *Backend) StartMining() bool {
 		cycles := 0
 		nhr := util.GetNetHash()
 		th := util.GetTipHeight()
+		var avgEarning float64
 		continueLoop := true
 		for continueLoop {
 			cycles++
@@ -113,8 +114,11 @@ func (m *Backend) StartMining() bool {
 
 			m.runtime.Events.Emit("networkHashRate", fmt.Sprintf("%0.2f %s", netHash, hashrateUnit))
 
-			coinsPerDay := util.GetCoinsPerDay(th)
-			avgEarning := float64(hr) / float64(nhr) * float64(coinsPerDay)
+			// Avoids wrong estimates when the backend is unavailable
+			if th != 0 && nhr != 0 {
+				coinsPerDay := util.GetCoinsPerDay(th)
+				avgEarning = float64(hr) / float64(nhr) * float64(coinsPerDay)
+			}
 
 			m.runtime.Events.Emit("avgEarnings", fmt.Sprintf("%0.2f VTC", avgEarning))
 

